apache/zeppelin
# Enhanced resource pool serialization

### What is this PR for?

This is a step towards creating a REST endpoint from resource pools  https://issues.apache.org/jira/browse/ZEPPELIN-699

It allows resources to be fully serializable so that the object that they contain can be passed around using thrift.
### What type of PR is it?

Improvement
### Todos
- [X] - Add serialization for rest endpoint
### What is the Jira issue?

Partially implements this:  https://issues.apache.org/jira/browse/ZEPPELIN-699

Note, that this pull request is dependent upon this one:  
https://github.com/apache/incubator-zeppelin/pull/781

Before that change is made, this will not compile.
### How should this be tested?

There is a test class "org.apache.zeppelin.ResourceSerializerTest", but with no front-end, there isn't an easy manual way to test this.

However, these steps will work:
1. Start Zeppelin
2. Run a paragraph.
3.  Inspect "org.apache.zeppelin.resource.ResourcePoolUtils.getAllResources()".  You will see that this paragraph's results are included in this resource pool.
### Screenshots (if appropriate)
### Questions:
- Does the licenses files need update? No
- Is there breaking changes for older versions? No
- Does this needs documentation?  No.
